1-foundations/2-control-flows-and-functions/foundations-s2.ipynb

```diff
@@ -510,128 +705,465 @@
+   "id": "norman-flood",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "This is a test function, with no input parameters.\n"
+     ]
+    }
+   ],
+   "source": [
+    "## Calling a function\n",
+    "\n",
+    "test_func()        ## remember str.upper() and str.lower() functions?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "id": "processed-zoning",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Function with one input parameter\n",
+    "\n",
+    "def what_day_is_it(day):\n",
+    "    print(\"Today is {}\".format(day))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "id": "together-karen",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Today is Monday\n"
+     ]
+    }
+   ],
+   "source": [
+    "what_day_is_it(\"Monday\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "id": "responsible-association",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Today is Garbage\n"
+     ]
+    }
+   ],
+   "source": [
+    "## It can take whatever value you want\n",
+    "\n",
+    "what_day_is_it(\"Garbage\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "id": "ed094af1-1f84-47f6-a1ae-f38a8c6324d6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## If you want to add some validations, you can do that in the following way\n",
+    "\n",
+    "def what_day_is_it(day):\n",
+    "    \n",
+    "    if day in ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']:\n",
+    "        \n",
+    "        print(\"Today is {}\".format(day))\n",
+    "        \n",
+    "    else:\n",
+    "        \n",
+    "        print(\"Invalid day!!\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "id": "548d6bdf-826a-4514-b355-993687773d06",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Today is Tuesday\n"
+     ]
+    }
+   ],
+   "source": [
+    "what_day_is_it(\"Tuesday\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "id": "45e6b8b5-a1e5-499f-aa10-efcf267a9dcd",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Invalid day!!\n"
+     ]
+    }
+   ],
+   "source": [
+    "## Now check the response\n",
+    "\n",
+    "what_day_is_it(\"Garbage\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "id": "assured-qatar",
+   "metadata": {},
+   "outputs": [],
    "source": [
     "## Function with a parameter with default value\n",
     "\n",
     "def what_day_is_it(day = \"Thursday\"):\n",
+    "    \n",
     "    print(\"Today is {}\".format(day))"
    ]
```

Comment:
Maybe first show what happens if someone use `what_day_is_it()` without parameters before there is a default value. Good learning opportunity to teach participants on how to read error messages when forgetting to pass parameters.

Then after you introduced default value `what_day_is_it(day = "Thursday"):` and show how it works, show what happens if you pass too many parameters

Reply:
Good point!